tracer/src/Datadog.Trace/Agent/TraceSamplers/RareSampler.cs

```diff
@@ -30,7 +36,16 @@ internal class RareSampler : ITraceChunkSampler
         /// <returns>true when a rare span is found, false otherwise</returns>
         public bool Sample(ArraySegment<Span> trace)
         {
-            if (SamplingHelpers.IsKeptBySamplingPriority(trace))
+            if (!IsEnabled)
```

Comment:
While you're in here, do you mind renaming the parameter `trace` to `traceChunk`? We're trying to make it clear that these are chunks everywhere, not necessarily full traces.

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3360/commits/e47895afa5fb745eeb754955307a65fae9db10df